ledger/acctupdates.go

```diff
@@ -1971,6 +1974,61 @@ func (au *accountUpdates) commitRound(offset uint64, dbRound basics.Round, lookb
 
 }
 
+// compactDeltas takes an arary of account map deltas ( one array entry per round ), and corresponding creatables array, and compact the arrays into a single
+// map that contains all the account deltas changes. While doing that, the function eliminate any intermediate account changes.
+// The function returns a map that could be the same as the input map, and therefore both the outputs of this function as well as the inputs to this function
+// should be kept read-only ( unless we don't care about the correctness of the data anymore ).
```

Comment:
right now, the input is already a read-only. The reason it's needed is because in the case of `if len(accountDeltas) == 1 && len(creatableDeltas) == 1`, we're copying the input maps into the output..
so, a change to either one of them would change the other.

This is not a real issue, since we don't modify either of these.